### PR TITLE
show, update, delete

### PR DESCRIPTION
**Pull request by DiptoSarkar182**
Added show, update and delete for category, sub category and sub category quiz.

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -23,9 +23,41 @@ def create
     end
   end
 
+  def show_category
+    result = Category.show_category(params[:category_id])
+    if result[:status] == :ok
+      render json: result[:data], status: :ok
+    else
+      render json: { message: result[:message] }, status: result[:status]
+    end
```
Comment (JakariaJishan) [inline app/controllers/categories_controller.rb:32]: I think it's fine to query here directly for show action

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -23,9 +23,41 @@ def create
...
+  end
+
+  def delete_category
+    result = Category.delete_category(params[:category_id])
+    if result[:status] == :ok
+      render json: { message: result[:message] }, status: :ok
+    else
+      render json: { message: result[:message] }, status: result[:status]
+    end
+  end
```
Comment (JakariaJishan) [inline app/controllers/categories_controller.rb:52]: I don't think that there is any necessity of separating methods at model level for normal crud action

File: app/controllers/sub_categories_controller.rb [hunk 3]
```diff
@@ -23,6 +23,34 @@ def create
...
+  end
+
+  def delete_sub_category
+    result = SubCategory.delete_sub_category(params[:sub_category_id])
+    if result[:status] == :ok
+      render json: { message: result[:message] }, status: :ok
+    else
+      render json: { message: result[:message] }, status: result[:status]
+    end
+  end
```
Comment (JakariaJishan) [inline app/controllers/sub_categories_controller.rb:52]: same goes here

File: app/controllers/sub_category_quizzes_controller.rb [hunk 4]
```diff
@@ -30,4 +55,13 @@ def answer
...
   end
+
+  private
+  def sub_category_quiz_params
+    params.require(:sub_category_quiz).permit(:sub_category_id, :quiz_question, :correct_answer_index, quiz_options: [])
+  end
+
+  def update_sub_category_quiz_params
+    params.require(:sub_category_quiz).permit(:sub_category_quiz_id, :sub_category_id, :quiz_question, :correct_answer_index, quiz_options: [])
+  end
```
Comment (JakariaJishan) [inline app/controllers/sub_category_quizzes_controller.rb]: Do you need two strong params? I think it's better to merge extra params with strong params when needed

File: app/controllers/sub_category_quizzes_controller.rb [hunk 5]
```diff
@@ -30,4 +55,13 @@ def answer
...
 end
```
Comment (JakariaJishan) [inline app/controllers/sub_category_quizzes_controller.rb]: add extra empty line at the bottom

File: app/models/category.rb [hunk 6]
```diff
@@ -15,4 +15,40 @@ def self.create_category(category_params)
...
+    category = Category.find_by(id: category_id)
+    if category
+      if category.destroy
+        { status: :ok, message: "Category deleted successfully"}
+      else
+        { status: :unprocessable_entity, message: "Failed to delete category", errors: category.errors.full_messages }
+      end
+    else
+      { status: :not_found, message: "Category not found!" }
+    end
```
Comment (JakariaJishan) [inline app/models/category.rb]: You are nesting two levels deep if else condition.
`return { status::not_found, message: "Category not found!" } unless category.present?`
this is a good approach I think

File: app/models/sub_category.rb [hunk 7]
```diff
@@ -35,4 +35,45 @@ def self.create_sub_category(sub_category_params)
...
+    if sub_category
+      if sub_category.destroy
+        { status: :ok, message: "Sub category deleted successfully"}
+      else
+        { status: :unprocessable_entity, message: "Failed to delete sub category", errors: category.errors.full_messages }
+      end
+    else
+      { status: :not_found, message: "Sub category not found!" }
+    end
+  end
```
Comment (JakariaJishan) [inline app/models/sub_category.rb]: same goes here

File: app/models/sub_category_quiz.rb [hunk 8]
```diff
@@ -2,11 +2,62 @@ class SubCategoryQuiz < ApplicationRecord
...
+    sub_category_quiz = SubCategoryQuiz.find_by(id: sub_category_quiz_id)
+    if sub_category_quiz
+      if sub_category_quiz.destroy
+        { status: :ok, message: "Sub category quiz deleted successfully"}
+      else
+        { status: :unprocessable_entity, message: "Failed to delete sub category quiz", errors: sub_category_quiz.errors.full_messages }
+      end
+    else
+      { status: :not_found, message: "Sub category quiz not found!" }
+    end
```
Comment (JakariaJishan) [inline app/models/sub_category_quiz.rb]: same here

**Review by JakariaJishan — CHANGES_REQUESTED** [review]
Please check all the comments

File: app/controllers/sub_category_quizzes_controller.rb (hunk 4 above)
Comment (DiptoSarkar182) [reply app/controllers/sub_category_quizzes_controller.rb]: Added extra params.

File: app/controllers/sub_category_quizzes_controller.rb (hunk 5 above)
Comment (DiptoSarkar182) [reply app/controllers/sub_category_quizzes_controller.rb]: Added extra line at the bottom.

File: app/models/category.rb (hunk 6 above)
Comment (DiptoSarkar182) [reply app/models/category.rb]: Added the suggested code.

File: app/models/sub_category.rb (hunk 7 above)
Comment (DiptoSarkar182) [reply app/models/sub_category.rb]: Added the suggested code.

File: app/models/sub_category_quiz.rb (hunk 8 above)
Comment (DiptoSarkar182) [reply app/models/sub_category_quiz.rb]: Added the suggested code.

**Review by JakariaJishan — APPROVED** [review]
Approved. It's time to merge.

**Pull request merged**